### PR TITLE
Sh/update

### PR DESCRIPTION
**Pull request by susuhahnml**
- Updated the License
- Updated the documentation with a nicer/simpler template

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2021 Susana Hahn
+Copyright (c) 2024 Potassco
```
Comment (rkaminsk) [inline LICENSE:3]: Could also be replaced by `init.py`.

File: README.md [hunk 2]
```diff
@@ -1,16 +1,22 @@
 # fillname
 
-This project template is configured to ease collaboration. Linters, formatters,
-and actions are already configured and ready to use.
```
Comment (rkaminsk) [inline README.md:3]: too many blank lines

File: README.md [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+
+>    Remove this block after following the instructions below to use the template
+>
+>    This project template is configured to ease collaboration. Linters, formatters,
+>    and actions are already configured and ready to use.
+>
+>    To use the project template, run the `init.py` script to give the project a
+>    name and some metadata. The script can then be removed. Then adjust the
+>    `setup.cfg` file as needed.
+
```
Comment (rkaminsk) [inline README.md:13]: too many blank lines

File: README.md [hunk 4]
```diff
@@ -25,14 +31,35 @@ To improve code quality, we run linters, type checkers, and unit tests. The
...
+```bash
+nox -s session_name
+```
+
+- See the list of sessions
+
+```bash
+nox -l
+```
+
```
Comment (rkaminsk) [inline README.md:60]: too many blank lines

File: README.md [hunk 5]
```diff
@@ -25,14 +31,35 @@ To improve code quality, we run linters, type checkers, and unit tests. The
 tools can be run using [nox]. We recommend installing nox using [pipx] to have
 it available globally:
 
+### Install [nox]
```
Comment (rkaminsk) [inline README.md:34]: The heading get a bit too fine-grained. What about keeping it simple and create one big bash code fence:

### Nox

```bash
# install
python -m pip install pipx
python -m pipx install nox

# run all sessions
nox

# list all sessions
nox -l

# run individual sessions
nox -s session_name
```

File: doc/_static/css/custom.css [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+.sidebar-brand-text {
+  text-align: center !important;
+  /* color: #004C6E !important; */
+  align-items: center;
+  display: inline-flex;
+}
+
+.sidebar-brand {
+  flex-direction: row !important;
+}
```
Comment (rkaminsk) [inline doc/_static/css/custom.css:34]: Looks like the newline checker did not run. :smile:

File: doc/conf.py [hunk 7]
```diff
@@ -9,7 +9,10 @@
 import configparser
 import datetime
 import os
+import sys
 
+sys.path.insert(0, os.path.abspath('..'))
```
Comment (rkaminsk) [inline doc/conf.py:14]: This looks ugly. Why do we need this?

File: doc/conf.py [hunk 8]
```diff
@@ -36,13 +39,20 @@
...
+    'sphinx.ext.napoleon',
+    'sphinx.ext.autosectionlabel',
+    'sphinx.ext.intersphinx',
     "sphinx.ext.autosummary",
-    "sphinx.ext.napoleon",
-    "sphinx.ext.autosectionlabel",
-    "sphinx.ext.intersphinx",
+    "sphinx_copybutton",
+    "myst_parser",
+    # "myst-parser"  # To write markdown documentation
```
Comment (rkaminsk) [inline doc/conf.py:55]: remove comment

File: doc/content/installation.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+You can check a successfull instalaltion by running
+
+```shell
+fillname -h
+```
+
+## Installing with pip 
+
+
+The python fillname package can be found [here](https://pypi.org/project/fillname/).
```
Comment (rkaminsk) [inline doc/content/installation.md:14]: The `init.py` script should probably also replace fillname here and in other documentation files.

File: doc/content/quickstart.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# Quick start
+
+A simple explanation on how to use the system
```
Comment (rkaminsk) [inline doc/content/quickstart.md:3]: period

File: doc/content/quickstart.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+```shell
+fillname -h
+```
+
+```{tip} 
+
+Tips on how to use the system
```
Comment (rkaminsk) [inline doc/content/quickstart.md:11]: period.

File: noxfile.py [hunk 12]
```diff
@@ -13,6 +13,9 @@
 
 @nox.session
 def format(session):
+    """
+    Formats the files to remove errors
```
Comment (rkaminsk) [inline noxfile.py:17]: please add periods at the end of docstrings.

Here I would also just write `Autoformat source files.`

File: noxfile.py [hunk 13]
```diff
@@ -53,28 +59,41 @@ def doc(session):
     session.install("-e", ".[doc]")
     session.cd("doc")
     session.run("sphinx-build", "-M", target, ".", "_build", *options)
+    session.run("open", "_build/html/index.html")
```
Comment (rkaminsk) [inline noxfile.py:62]: Please, let's not do this by default. I would not mind an option.

File: noxfile.py [hunk 14]
```diff
@@ -53,28 +59,41 @@ def doc(session):
...
 
 
 @nox.session
 def lint_flake8(session):
+    """
+    Runs flake8 linter
```
Comment (rkaminsk) [inline noxfile.py:68]: Run flake8 linter.

File: noxfile.py [hunk 15]
```diff
@@ -53,28 +59,41 @@ def doc(session):
...
+    """
     session.install("-e", ".[lint_flake8]")
     session.run("flake8", "src", "tests")
 
 
 @nox.session
 def lint_pylint(session):
+    """
+    Runs pylint
```
Comment (rkaminsk) [inline noxfile.py:77]: Run pylint.

File: noxfile.py [hunk 16]
```diff
@@ -53,28 +59,41 @@ def doc(session):
...
+    """
     session.install("-e", ".[lint_pylint]")
     session.run("pylint", "fillname", "tests")
 
 
 @nox.session
 def typecheck(session):
+    """
+    Typechecks the code using mypy
```
Comment (rkaminsk) [inline noxfile.py:86]: Typecheck code using mypy.

File: noxfile.py [hunk 17]
```diff
@@ -53,28 +59,41 @@ def doc(session):
...
+    """
     session.install("-e", ".[typecheck]")
     session.run("mypy", "-p", "fillname", "-p", "tests")
 
 
 @nox.session(python=PYTHON_VERSIONS)
 def test(session):
+    """
+    Runs the tests
```
Comment (rkaminsk) [inline noxfile.py:95]: Run tests.

File: noxfile.py [hunk 18]
```diff
@@ -85,4 +104,7 @@ def test(session):
 
 @nox.session
 def dev(session):
+    """
+    Creates a development environment
```
Comment (rkaminsk) [inline noxfile.py:108]: Create development session.

File: setup.cfg [hunk 19]
```diff
@@ -38,8 +38,11 @@ test =
     coverage
 doc =
     sphinx
-    sphinx_rtd_theme
+    furo
     nbsphinx
+    sphinx_copybutton
+    sphinx-togglebutton
```
Comment (rkaminsk) [inline setup.cfg:44]: the togglebutton extension does not seem to be used?

File: noxfile.py [hunk 20]
```diff
@@ -44,6 +47,9 @@ def format(session):
 
 @nox.session
 def doc(session):
+    """
+    Builds and opens the documentation
```
Comment (rkaminsk) [inline noxfile.py:51]: Build documentation.

File: doc/content/installation.md [hunk 21]
```diff
@@ -0,0 +1,41 @@
+# Installation
+
+fillname requires a Python version between 3.8 and 3.10 (recomend 3.10)
```
Comment (rkaminsk) [inline doc/content/installation.md:3]: fillname requires Python 3.8+. We recommend version 3.10.

**Review by rkaminsk — CHANGES_REQUESTED** [review]
Some comments. Currently, the init.py script is broken. It should be adjusted to replace fillname and maybe some metadata in the documentation.

The documentation also needs a spellcheck. I'll do one once the technical stuff is in place.

File: README.md (hunk 5 above)
Comment (susuhahnml) [reply README.md:34]: I added what you suggest. I would even suggest that the rest of the text goes into the doc string of the sessions to simplify the README

File: doc/conf.py (hunk 7 above)
Comment (susuhahnml) [reply doc/conf.py:14]: Removed :)

File: noxfile.py (hunk 13 above)
Comment (susuhahnml) [reply noxfile.py:62]: I added this and other options

File: README.md (hunk 5 above)
Comment (rkaminsk) [reply README.md]: I added back one paragraph, which I think is important.

**Comment by rkaminsk** [comment]
I tried to fix the init.py script. It should still be actually run to see if it works now.

**Comment by susuhahnml** [comment]
The docstrings only show the first line in the `nox -l` command, that is why I had them as a single line. Could not find a way to show it all

File: init.py [hunk 22]
```diff
@@ -43,28 +43,18 @@ def replace(filepath):
...
-        replace(filepath)
+    dirs = [os.path.join("src", "fillname"), "tests", "doc"]
+    files = ["setup.cfg", "noxfile.py", "README.md", "LICENSE", ".pre-commit-config.yaml", ".coveragerc"]
 
-    for rootpath in directories_to_replace:
+    for rootpath in dirs:
         for dirpath, _, filenames in os.walk(rootpath):
             for filename in filenames:
-                if not filename.endswith(".py"):
+                if not filename.endswith(".py") and not filename.endswith(".md"):
```
Comment (susuhahnml) [inline init.py:59]: @rkaminsk  I did use fillname in the documentation md (for the install and use). Is there a reason why we are skipping them?

Comment (rkaminsk) [reply init.py:59]: We don't. Only other files are skipped.

**Comment by rkaminsk** [comment]
> The docstrings only show the first line in the `nox -l` command, that is why I had them as a single line. Could not find a way to show it all

I think it's okay not to show everything. Otherwise, it just gets too ugly.

**Comment by susuhahnml** [comment]
Good for me to merge :)

**Comment by rkaminsk** [comment]
I have the feeling that the `init.py` script is completely untested. I did not run it either. At least one of us should do it. :wink:

**Comment by susuhahnml** [comment]
I can do it

**Review by rkaminsk — APPROVED** [review]
Fine to merge from my side.

**Pull request merged**